### PR TITLE
[Feature] Add the frozen function for Swin Transformer model

### PR DESCRIPTION
**Pull request by fangxu622**
## Motivation

Adding feature request. It provides frozen function for swin transformer by adding  add frozen(bool) parameter .

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by mzr1996** [comment]
Thanks for your contribution, we will review it soon.

File: mmcls/models/backbones/swin_transformer.py [hunk 1]
```diff
@@ -425,3 +430,26 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, *args,
...
+    def train(self, mode=True):
+        super(SwinTransformer, self).train(mode)
+        self._freeze_stages()
+        if mode and self.norm_eval:
+            for m in self.modules():
+                # trick: eval have effect on BatchNorm only
+                if isinstance(m, _BatchNorm):
+                    m.eval()
+                if isinstance(m, nn.modules.LayerNorm):
+                    m.eval()
```
Comment (mzr1996) [inline mmcls/models/backbones/swin_transformer.py:455]: Does `LayerNorm` need to turn to evaluation mode? Unlike `BatchNorm`, `LayerNorm` should have the same behavior in both training mode and eval mode.

Comment (fangxu622) [reply mmcls/models/backbones/swin_transformer.py:455]: It also puzzles me.  I write this reference https://github.com/open-mmlab/mmclassification/blob/9d9dce69ad1d7911c87c69e30bce4a928f4dcd94/mmcls/models/backbones/resnet.py#L632 .

**Review by mzr1996 — CHANGES_REQUESTED** [review]

**Review by mzr1996 — APPROVED** [review]
LGTM
